modules/core/src/main/java/org/apache/ignite/internal/processors/odbc/ClientListenerMetrics.java

```diff
@@ -0,0 +1,157 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.ignite.internal.processors.odbc;
+
+import java.util.HashMap;
+import java.util.Map;
+import org.apache.ignite.internal.GridKernalContext;
+import org.apache.ignite.internal.processors.metric.MetricRegistry;
+import org.apache.ignite.internal.processors.metric.impl.IntMetricImpl;
+import org.apache.ignite.internal.processors.metric.impl.MetricUtils;
+
+import static org.apache.ignite.internal.processors.metric.GridMetricManager.CLIENT_METRICS;
+import static org.apache.ignite.internal.processors.odbc.ClientListenerNioListener.JDBC_CLIENT;
+import static org.apache.ignite.internal.processors.odbc.ClientListenerNioListener.ODBC_CLIENT;
+import static org.apache.ignite.internal.processors.odbc.ClientListenerNioListener.THIN_CLIENT;
+
+/**
+ * Client listener metrics.
+ */
+public class ClientListenerMetrics {
+    /** Connections metric label. */
+    public static final String METRIC_CONNECTIONS = "connections";
+
+    /** Handshakes rejected by timeout metric label. */
+    public static final String METRIC_REJECTED_TIMEOUT =
+            MetricUtils.metricName(METRIC_CONNECTIONS, "rejectedByTimeout");
+
+    /** Handshakes rejected by authentication metric label. */
+    public static final String METRIC_REJECTED_AUTHENTICATION =
+            MetricUtils.metricName(METRIC_CONNECTIONS, "rejectedAuthentication");
+
+    /** Total number of rejected handshakes. */
+    public static final String METRIC_REJECTED_TOTAL =
+            MetricUtils.metricName(METRIC_CONNECTIONS, "rejectedTotal");
+
+    /** Rejected by timeout. */
+    private final IntMetricImpl rejectedTimeout;
+
+    /** Rejected by authentication. */
+    private final IntMetricImpl rejectedAuth;
+
+    /** Total number of rejected connections. */
+    private final IntMetricImpl rejectedTotal;
+
+    /** Connections accepted. */
+    private final Map<Byte, IntMetricImpl> accepted;
```

Comment:
Nitpick: Map can be replaced with an array, here and below.

Reply:
Implemented